src/librbd/AsyncResizeRequest.cc

```diff
@@ -119,8 +119,7 @@ void AsyncResizeRequest::send_grow_object_map() {
   bool object_map_enabled = true;
   {
     RWLock::RLocker l(m_image_ctx.owner_lock);
-    RWLock::RLocker l2(m_image_ctx.md_lock);
-    if (m_image_ctx.object_map == NULL) {
+    if (!m_image_ctx.object_map.enabled()) {
```

Comment:
Might want to be holding snap_lock where md_lock was previously held.  Otherwise an image refresh could change the state of the object map.  Similar issue is in a few more spots where md_lock was removed.